service/src/main/java/bio/terra/landingzone/service/landingzone/azure/LandingZoneService.java

```diff
@@ -111,6 +115,44 @@ public void deleteLandingZone(String landingZoneId) {
     throw new LandingZoneDeleteNotImplemented("Delete operation is not implemented");
   }
 
+  public Map<String, List<LandingZoneResource>> listGeneralResourcesWithPurposes(
```

Comment:
At the service level you could combine the results from the library. As mentioned earlier you can use the listResourcesByPurpose and listSubnetsByPurpose, to create a list of all the resources grouped by purpose.  Introducing two methods here and pushing the combination to the controller or any wrapper above the service, dilutes the value of the service component and potentially complicates testing.

Reply:
Done